### PR TITLE
Brand logo CSS improvements

### LINKED ISSUE #50: Improve layout styles to improve compatability with different logo sizes
When following the instructions in the docs to add a custom logo, the current layout styles are not very accommodating to different logo sizes or proportions resulting in the logo often displaying badly and requiring manual CSS adjustments in order to make it work.

Can I suggest adjusting the layout styles to set maximum widths and heights for any logo image, and to also automatically adjust the height of the logo grid cell appropriately, rather than setting it to fixed height in a CSS variable? This would then allow developers to much more easily add a custom logo.

e.g. currently, you end up with something like this:

But if we set the height of the cell to `auto` instead of `calc(var(--sizing-app-brand-height) + (var(--spacing-app-brand) * 2))` and set the `max-width` and `max-height` of any logo image something like `300px` and `150px`, we get this:

This same CSS works with other logo shapes without any adjustment required, e.g.: 

This would also make it possible to make the logo URL a simple config option, rather than requiring a template override.

Or have I missed something? Is there a particular reason it has been done the way it has?

### PR DESCRIPTION
**Pull request by purplespider**
CSS now automatically scales logo size for compatibility with different logo image shapes and sizes. Logo image tag can now be added to `AppBrand.ss` and no CSS adjustments should be necessary.

Tested with square, portrait and landscape orientated logos, in PNG and SVG formats. In all supported SilverStripe browsers including IE11.

Resolves #50

### CODE REVIEW

File: client/src/styles/brand.scss
Comment (Cheddam) [inline client/src/styles/brand.scss]: What purpose does this property serve? I tried removing it in my testing, and it didn't seem to have an impact - but perhaps it serves an edgecase I wasn't replicating?

**Review by Cheddam — APPROVED** [review]
Nailed it! One minor query - otherwise happy to merge.

File: client/src/styles/brand.scss
Comment (purplespider) [reply client/src/styles/brand.scss]: Hmm, you're right! Doesn't seem to have an effect, and I can't actually remember what it was for. It was possibly to solve alignment issues before I decided to move the `app-brand__logo` class from a wrapper `div` to the `img` tag itself.

Comment (Cheddam) [reply client/src/styles/brand.scss]: Sweet, makes sense - would you mind dropping it out? Once that's pushed we're good to merge 👍

Comment (purplespider) [reply client/src/styles/brand.scss]: Done!

**Comment by ScopeyNZ** [comment]
This can probably be a bugfix and can be targetted at 4.1.  I've noticed that you've created this pull request from one of the "core branches" rather than a branch specifically intended for this pull request. This can cause problems if we ask you to target this PR at a different branch. No action is needed at the moment, this is for your information.

In the future please create a new branch for each individual pull request. See our [guide about contributing PRs](https://docs.silverstripe.org/en/4/contributing/code/#step-by-step-from-forking-to-sending-the-pull-request).

Thanks for the PR @purplespider! @Cheddam I'll let you decide whether you want it in a patch branch. I think it might as well be, but it might just be easier to squash/merge this and cherry-pick the fix into the patch branch?

**Comment by Cheddam** [comment]
We can cherry-pick it, yep. Thanks @purplespider!

**Pull request merged**
